### DISCUSSION TITLE
ERROR Unit name: field must be populated ...

### DISCUSSION THREAD

**Discussion by AndyB117** [davetcc/tcMenu#447] [API & Remote]
There is an error when you try to specify "Unit name" with character other than letter or digit.
Even if character is ASCII, for example "%". And it is not saved.
"%" as unit name is often used choice.
But if we change Unit name in generated code, all worked. We can use %, °, °C and so on.
How to bypass this limitation?
tcMenu v 4.2.1

**Comment by davetcc** [comment]
This is a bug in 4.2.1. Raised issue #448 

For now, you can edit that particular value in the underlying emf project file, see the [CLI instructions](https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/tcmenu-cli-workflow/) for more on that. It will be fixed in the next release.

**Comment by davetcc** [comment]
Fixed in the latest patch release
